src/db/shortcuts.ts

```diff
@@ -58,36 +73,60 @@ type LimitedLateralOption = SQLFragmentMap | undefined;
 type FullLateralOption = LimitedLateralOption | SQLFragment<any>;
 type LateralOption<
   C extends ColumnsOption<Table>,
-  E extends ExtrasOption<Table>,
-> =
-  undefined extends C ? undefined extends E ? FullLateralOption : LimitedLateralOption : LimitedLateralOption;
-
-export interface ReturningOptionsForTable<T extends Table, C extends ColumnsOption<T>, E extends ExtrasOption<T>> {
+  E extends ExtrasOption<Table>
+> = undefined extends C
+  ? undefined extends E
+    ? FullLateralOption
+    : LimitedLateralOption
+  : LimitedLateralOption;
+
+export interface ReturningOptionsForTable<
+  T extends Table,
+  C extends ColumnsOption<T>,
+  E extends ExtrasOption<T>
+> {
   returning?: C;
   extras?: E;
-};
-
-type ReturningTypeForTable<T extends Table, C extends ColumnsOption<T>, E extends ExtrasOption<T>> =
-  (undefined extends C ? JSONSelectableForTable<T> :
-    C extends ColumnForTable<T>[] ? JSONOnlyColsForTable<T, C> :
-    never) &
-  (undefined extends E ? {} :
-    E extends SQLFragmentOrColumnMap<T> ? ExtrasResult<T, E> :
-    never);
-
+}
 
-function SQLForColumnsOfTable(columns: readonly Column[] | undefined, table: Table) {
-  return columns === undefined ? sql`to_jsonb(${table}.*)` :
-    sql`jsonb_build_object(${mapWithSeparator(columns, sql`, `, c => sql`${param(c)}::text, ${c}`)})`;
+type ReturningTypeForTable<
+  T extends Table,
+  C extends ColumnsOption<T>,
+  E extends ExtrasOption<T>
+> = (undefined extends C
+  ? JSONSelectableForTable<T>
+  : C extends ColumnForTable<T>[]
+  ? JSONOnlyColsForTable<T, C>
+  : never) &
+  (undefined extends E
+    ? {}
+    : E extends SQLFragmentOrColumnMap<T>
+    ? ExtrasResult<T, E>
+    : never);
+
+function SQLForColumnsOfTable(
+  columns: readonly Column[] | undefined,
+  table: Table
+) {
+  return columns === undefined
+    ? sql`to_jsonb(${table}.*)`
```

Comment:
so I think for me the main next thing to understand (and I write this because you may have grappled with it already?) is the relationship between this change and this call to `to_jsonb`. so as I understand it, right now at least in the `.select`/etc path (not the `.sql` path) is that postgres returns a big old jsonb object, which is then converted with `to_jsonb`, and then there is a little post processing. so for the initial decimal support, we changed the function that it uses to parse jsonb objects. in this case we are changing an entirely different code path...at least in the select path, are the changes taking effect after the `to_jsonb` call? I will spend some time later tracing through what actually happens and try to tie it all together, but I just want to make sure that these two parts play nicely together because I think there is a chance of some mismatched conversions etc